### PR TITLE
[rocm-3.10.x] add bwd upsampling clear check

### PR DESCRIPTION
**Pull request by carlushuang**
create PR same as #566 to for `rocm-3.10.x` branch

### CODE REVIEW

File: src/conv/invokers/impl_gemm_dynamic.cpp [hunk 1]
```diff
@@ -304,7 +304,16 @@ InvokerFactory MakeImplGemmDynamicForward1x1InvokerFactory(const ConvolutionCont
 InvokerFactory MakeImplGemmDynamicBackwardDataInvokerFactory(const ConvolutionContext& ctx)
 {
     const auto& conv_problem = ctx.conv_problem;
-    return [conv_problem](const std::vector<Kernel>& kernels) {
+    int stride_h             = conv_problem.GetInHeight() > 1 ? conv_problem.GetKernelStrideH() : 1;
```
Comment (asroy) [inline src/conv/invokers/impl_gemm_dynamic.cpp:307]: Is GetInHeight() input (Hi) or really just output image height (Ho)? I know ctx swap them internally.

Ho should be used here for adjusting stride_h.

I think you need some comment here to explain if Hi/Wo get swapped, and that you are adjust stride_h based on Ho, etc

Comment (asroy) [inline src/conv/invokers/impl_gemm_dynamic.cpp:307]: Is GetInHeight() input (Hi) or really just output image height (Ho)? I know ctx swap them internally.

Ho should be used here for adjusting stride_h.

I think you need some comment here to explain if Hi/Wo get swapped, and that you are adjusting stride and dilation based on Ho, etc

**Review by asroy — CHANGES_REQUESTED** [review]

File: src/conv/invokers/impl_gemm_dynamic.cpp (hunk 1 above)
Comment (carlushuang) [reply src/conv/invokers/impl_gemm_dynamic.cpp:307]: Yes. In backward-data context, the GetInHeight() is indeed the output-gradient tensor hight. This is internally swapped in ctx. Let me add some comment about this

**Review by asroy — APPROVED** [review]

**Comment by asroy** [comment]
@daniellowell This PR is doing the same fix as #566, but for ```rocm-3.10.x```. But since you were able to cherry pick #566 into ```rocm-3.10.x-staging```, do we still need this PR?

**Comment by daniellowell** [comment]
@carlushuang I cherry-picked the 566 PR into rocm-3.10.x-staging.

**Pull request closed without merging**